context.go

```diff
@@ -43,41 +47,53 @@ type RequestContext struct {
 }
 
 func initReqCtx(rc *RequestContext, headers *corev3.HeaderMap) error {
+
+	allHeaders, err := genHeaders(headers)
+	if err != nil {
+		return err
+	}
+
 	rc.Started = time.Now()
 	rc.Duration = 0
-	rc.Headers = make(map[string][]string)
 
 	// for custom data between phases
 	rc.data = make(map[string]any)
 
 	// for stream phase responses (convenience)
 	rc.ResetPhase()
 
-	for _, h := range headers.Headers {
-		switch h.Key {
+	// string and byte header processing
+	rc.ByteHeaders = allHeaders.ByteHeaders
+	rc.Headers = make(map[string][]string)
+	for k, v := range allHeaders.Headers {
+		switch k {
 		case ":scheme":
-			rc.Scheme = h.Value
+			rc.Scheme = v[0]
 
 		case ":authority":
-			rc.Authority = h.Value
+			rc.Authority = v[0]
 
 		case ":method":
-			rc.Method = h.Value
+			rc.Method = v[0]
 
 		case ":path":
-			rc.Path = strings.Split(h.Value, "?")[0]
+			rc.Path = strings.Split(v[0], "?")[0]
 
 		case "x-request-id":
-			rc.RequestId = h.Value
+			rc.RequestId = v[0]
 
 		default:
-			rc.Headers[h.Key] = strings.Split(h.Value, ",")
+			rc.Headers[k] = v
 		}
 	}
 
 	return nil
 }
 
+func (rc *RequestContext) GetAllHeaders() AllHeaders {
```

Comment:
`AllHeaders`?, the`GetXX` is not very go style

Reply:
Updated